### PR TITLE
Add opts.request

### PR DESCRIPTION
**Pull request by es128**
Resolves gh-51

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ var path = require('path');
 var rename = require('gulp-rename');
 var through = require('through2');
 var vfs = require('vinyl-fs');
+var objectAssign = require('object-assign');
```
Comment (shinnn) [inline index.js]: Keep A- Z order.

```js
// ...
var isUrlSuperb = require('is-url-superb');
var objectAssign = require('object-assign');
var path = require('path');
// ...
```

Comment (es128) [inline index.js]: 252e329

**Comment by shinnn** [comment]
:+1: LGTM

**Comment by kevva** [comment]
I would prefer just keeping it in the regular opts.

**Comment by es128** [comment]
What do you mean? Pass *all* your `download` opts to `got` and expect it to ignore the irrelevant ones?

**Comment by kevva** [comment]
Yeah, exactly.

**Comment by es128** [comment]
Alright, but that does mean that if `got` or `http`/`https` ever add a conflicting option down the road you'll likely need to make breaking API changes.

**Comment by kevva** [comment]
No, we can just delete them from the `opts` object and set them in a new object or something.

**Comment by es128** [comment]
That wouldn't be sufficient if you have users relying on your version of the option and others who want to start using the downstream versions of that option.

But it's ok, there's a reasonably good chance this will never happen.

**Comment by kevva** [comment]
Ah, yes, that's true. I'll deal with that later if that'll happen.

**Comment by es128** [comment]
Alright, PR updated

**Comment by es128** [comment]
@kevva ping

**Comment by kevva** [comment]
Sorry for the delay, could you just squash your commits :)?

**Comment by es128** [comment]
Sure, was just waiting to ensure there were no more changes. Done.

**Comment by kevva** [comment]
Great, thanks!

**Pull request merged**